### PR TITLE
Fix lings spending double the chemicals for reviving stasis (30->15)

### LINKED ISSUE #82394: Changeling Reviving Stasis costs 30 chemicals instead of 15
## Issue Summary

Changeling reviving stasis says in its tooltip that it costs 15 chemicals to use. I used it the other round and it cost 15 to go down and 15 to get back up. I found that odd, because why would it say 15 instead of 30? So, I took a look at the code and did some testing: there is actual explicit code for setting the cost to 0 to revive from the stasis, however there is an early signal being sent in the middle of the revival that causes the cost to be reset to default unintentionally early. I don't know how signals work, but I want to submit this bug report so I can reference it in the PR fix when I figure out signals.

## Reproduction:

Go into reviving stasis. See that you get charged 15 chems to go down, and 15 chems to get up. Take a look at the code in fakedeath.dm

### PR DESCRIPTION
**Pull request by Metekillot**
## About The Pull Request

Currently, reviving stasis is bugged to force lings to spend the chemical cost twice. This change refactors the fakedeath code to instead reset the chemical cost after the revival ability resolves, instead of resetting it in the middle of their revival before the chemical adjustment is resolved.
## Why It's Good For The Game

Fixes #82394 
## Changelog

Changelings now only have to pay the intended 15 chemicals for reviving stasis.
:cl: Bisar
fix: Adjusted the code for changeling reviving stasis to only make them spend chemicals once.
/:cl:

### CODE REVIEW

File: code/modules/antagonists/changeling/powers/fakedeath.dm
Comment (Metekillot) [inline code/modules/antagonists/changeling/powers/fakedeath.dm:1]: ignore the messed up fakedeath duration, i fixed it in a subsequent commit

File: code/modules/antagonists/changeling/powers/fakedeath.dm [hunk 1]
```diff
@@ -51,14 +50,21 @@
 /// Can be called mid-revival if the process is being cancelled
 /datum/action/changeling/fakedeath/proc/disable_revive(mob/living/changeling)
 	if(revive_ready)
-		chemical_cost = 15
-		revive_ready = FALSE
-		build_all_button_icons(UPDATE_BUTTON_NAME|UPDATE_BUTTON_ICON)
+		addtimer(CALLBACK(src, PROC_REF(reset_chemical_cost)), 2 DECISECONDS, TIMER_UNIQUE)
+		//a timer is used here because this needs to be done after the upstream logic
+		//of try_to_sting()
```
Comment (MrMelbert) [inline code/modules/antagonists/changeling/powers/fakedeath.dm:55]: Doesn't this mean you can spam click the button after reviving (in the 2 decisecond window, to go right back into revival stasis for free? (Not that I think it is extremely useful per se but still?)

Comment (MrMelbert) [reply code/modules/antagonists/changeling/powers/fakedeath.dm:55]: Or actually thinking about it more it's not going back into revival stasis, you can call revive() multiple times per stasis

**Comment by MrMelbert** [comment]
Since the problem is that we need to reset after `try_to_sting`, why not just move resetting chem cost to... after `try_to_sting`?
```dm
/datum/action/changeling/fakedeath/try_to_sting/disable_revive(mob/living/user, mob/living/target)
    . = ..()
    if(!. || !revive_ready)
        return
    reset_chemical_cost()
```

File: code/modules/antagonists/changeling/powers/fakedeath.dm (hunk 1 above)
Comment (Metekillot) [reply code/modules/antagonists/changeling/powers/fakedeath.dm:55]: Oh, I didn't think of that, I'll have to take another look.

Comment (san7890) [reply code/modules/antagonists/changeling/powers/fakedeath.dm:55]: you should really just use the `COOLDOWN_DECLARE` system or something

**Comment by Metekillot** [comment]
Thank you melbert!

**Pull request merged**